mcode/mcode-coverage-checker
# Steam 976/file upload

# Description
Issue: 976

This PR adds functionality to support a file upload component that connects to the state displayed by the (previously) PastFiles component (now UploadedFiles). This works via click-and-select and drag-and-drop. Note: some of the section headers and icons differ from what are in Alden's designs. These changes were discussed with him, but feel free to double check with him on anything that doesn't look right to you.

## (Feature) Demo/Screenshots

https://user-images.githubusercontent.com/5703736/199584868-200dcd2d-d4af-493a-bdaf-5c4869b7cdc9.mov

## Important Changes

The important changes take place in `src/components/FileUpload.js`, but the dummy data that used to live in PastFiles has been moved (temporarily) to be the default data in the recoil state file. 

Additional changes because of this PR include: 
- **Updates the `basename` to be mcode-coverage-checker, instead of the home page route. This means file upload is accessed via the `mcode-coverage-checker/file-upload` route, not just `file-upload`.**
- Updates the README to align with the project description used on the Github project metadata 
- Adds the following support libraries: 
  - [AutoAnimate](https://auto-animate.formkit.com/#usage-react) for easy animation of child components being added and removed from parent components 
  - [UUID](https://www.npmjs.com/package/uuid) for easy generation of unique ID's
  - [Recoil](https://recoiljs.org/) for a simple, global state management library
- Moves app styles to the `styles` directory.


## Testing Recommendations
- Ensure that drag and drop works for uploading files
- Ensure that click and select works for uploading files 
- Ensure that the file is reflected in the All Files section 
- Ensure that you can dismiss notifications using the `X` button
- Ensure that dismissing notifications doesn't delete the files in All Files.

# Checklists

**Submitter:**

@ACCT1 :
